testutil/node.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package testutil
+
+import (
+	"net"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	"github.com/ethereum/go-ethereum/p2p/simulations/adapters"
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/network"
+)
+
+// NodeConfigAtPo brute forces a node config to create a node that has an overlay address at the provided po in relation to the given baseaddr
+func NodeConfigAtPo(t *testing.T, baseaddr []byte, po int) *adapters.NodeConfig {
```

Comment:
I'd prefer if we don't add test code in the main binary... maybe rename this to `util_test.go` or something ending with `_test.go` ?

Reply:
we can't. either it sits in this package or it gets added to the binary. this functionality is super important in our test suite and should be accessible from any package. i don't have any solution for this

Reply:
I think that if this package is imported only in test files, it will not be included in the binary.

Reply:
I still don't understand why we need a `testutil` package, and why can't that live next to the test code. It doesn't appear to be used by anything else, but the retrieval tests.

Reply:
this is going to be used in the new stream package for certain tests. that's why i put it here so i dont need to rebase or clip this out of the directory just in a few days. that's all